retry redirect if 400

from https://github.com/dandi/dandi-cli/pull/725/checks?check_run_id=3135699904

```
_ test_download_000027_metadata_only[https://dandiarchive.org/dandiset/000027/draft] _
395
dandi/tests/test_download.py:65: in test_download_000027_metadata_only
396
    ret = download(url, tmpdir, get_assets=False)
397
dandi/download.py:59: in download
398
    parsed_url = parse_dandi_url(urls[0])
399
dandi/dandiarchive.py:378: in parse
400
    new_url = cls.follow_redirect(url)
401
dandi/dandiarchive.py:482: in follow_redirect
402
    f"Response for getting {url} to redirect returned {r.status_code}."
403
E   dandi.exceptions.FailedToConnectError: Response for getting https://dandiarchive.org/dandiset/000027/draft to redirect returned 400. Please verify that it is a URL related to dandiarchive and supported by dandi client
```

attn @satra -- may be smth to be tuned up on redirector.  I think this is not first/unique incident

--- a/dandi/dandiarchive.py
+++ b/dandi/dandiarchive.py
@@ -1,6 +1,7 @@
 from abc import ABC, abstractmethod
 from contextlib import contextmanager
 import re
+from time import sleep
 from typing import Iterator, Optional, Tuple
 from urllib.parse import unquote as urlunquote
 
@@ -474,18 +475,25 @@ class _dandi_url_parser:
 
     @staticmethod
     def follow_redirect(url):
-        r = requests.head(url, allow_redirects=True)
-        if r.status_code == 404:
-            raise NotFoundError(url)
-        elif r.status_code != 200:
-            raise FailedToConnectError(
-                f"Response for getting {url} to redirect returned {r.status_code}."
-                f" Please verify that it is a URL related to dandiarchive and"
-                f" supported by dandi client"
-            )
-        elif r.url != url:
-            return r.url
-        return url
+        i = 0
+        while True:
+            r = requests.head(url, allow_redirects=True)
+            if r.status_code == 404:
+                if i < 3:
+                    sleep(0.1 * 10 ** i)
+                    i += 1
+                    continue
+                else:
+                    raise NotFoundError(url)
+            elif r.status_code != 200:
+                raise FailedToConnectError(
+                    f"Response for getting {url} to redirect returned {r.status_code}."
+                    f" Please verify that it is a URL related to dandiarchive and"
+                    f" supported by dandi client"
+                )
+            elif r.url != url:
+                return r.url
+            return url
 
 
 # convenience binding

